test/e2e/specs/editor/various/block-hierarchy-navigation.spec.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+const COLUMNS_BLOCK = [
+	{
+		name: 'core/columns',
+		innerBlocks: [
+			{
+				name: 'core/column',
+				innerBlocks: [
+					{
+						name: 'core/paragraph',
+						attributes: { content: 'First column' },
+					},
+				],
+			},
+			{
+				name: 'core/column',
+			},
+			{
+				name: 'core/column',
+				innerBlocks: [
+					{
+						name: 'core/paragraph',
+						attributes: { content: 'Third column' },
+					},
+				],
+			},
+		],
+	},
+];
+
+test.describe( 'Navigating the block hierarchy', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should navigate using the list view sidebar', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.openDocumentSettingsSidebar();
+		await editor.insertBlock( { name: 'core/columns' } );
+		await editor.canvas.click(
+			'role=button[name="Two columns; equal split"i]'
+		);
+
+		// Open the block inserter.
+		await page.keyboard.press( 'ArrowDown' );
+		await page.keyboard.press( 'Enter' );
+
+		// Add a paragraph in the first column.
+		const paragraph = page.getByRole( 'option', { name: 'Paragraph' } );
+		await expect( paragraph ).toBeVisible();
+		await paragraph.click();
+		await page.keyboard.type( 'First column' );
+
+		await pageUtils.pressKeys( 'access+o' );
+
+		const listView = page.getByRole( 'treegrid', {
+			name: 'Block navigation structure',
+		} );
+
+		await expect( listView ).toBeVisible();
+		await listView
+			.getByRole( 'gridcell', { name: 'Columns', exact: true } )
+			.click();
+
+		// Tweak the columns count.
+		await page.getByRole( 'spinbutton', { name: 'Columns' } ).click();
+		await pageUtils.pressKeys( 'primary+a' );
+		await page.keyboard.type( '3' );
```

Comment:
Nit: Could we just use `.fill()` here? Also is this from the editor settings sidebar? Do you think it'd be easier to read if we scope this under it?

Reply:
Sure. I've updated the code.